docs/Onboarding Docs/01_Razor_Basics.md

```diff
@@ -0,0 +1,52 @@
+| --- | --- | --- | --- |
+| `.cshtml` | Razor View | Part of the MVC (Model-View-Controller) pattern, where the View is responsible for the presentation logic. Located within the Views folder of an MVC application and associated with a Controller. | Used in MVC applications for complex scenarios where separation of concerns is important. |
+| `.cshtml` | Razor Page | A page-based programming model that makes building web UI easier and more productive. Located within the Pages folder of a Razor Pages application and includes a `@page` directive at the top. | Used in Razor Pages applications for simpler scenarios where a full MVC model might be overkill. |
+| `.razor` | Razor Component (Blazor) | Used in Blazor, a framework for building interactive client-side web UI with .NET. Each `.razor` file is a self-contained component that can include both the markup and the processing logic. | Used in Blazor applications for building interactive client-side web UIs. |
+| `.vbhtml` | Razor View (VB.NET) | Part of the MVC (Model-View-Controller) pattern, where the View is responsible for the presentation logic. Located within the Views folder of an MVC application and associated with a Controller. | Used in older MVC applications written in VB.NET. |
+
+## Razor Editors: Legacy vs New
+
+| Aspect | Razor Legacy | Legacy .NET Core Razor Editor | New .NET Core Razor Editor |
+| --- | --- | --- | --- |
+| **Introduction** | Introduced with ASP.NET MVC 3. | Older Razor editor for ASP.NET Core projects. | Updated Razor editor introduced in Visual Studio 2019 version 16.8. |
+| **Usage** | Used in ASP.NET MVC and ASP.NET Web Pages applications. | Used for editing Razor views and pages in ASP.NET Core projects. | Used for editing Razor views and pages in ASP.NET Core projects. |
+| **Source code** | Closed source. | Closed source. | [Open source on GitHub](https://github.com/dotnet/razor/) |
+| **File Extensions** | `.cshtml` for C#, `.vbhtml` for VB.NET. | N/A | N/A |
+| **Functionality** | Creates dynamic web pages that combine HTML and server-side code. | Provides basic features like syntax highlighting and IntelliSense for Razor syntax. | Provides improved functionality and performance, including better IntelliSense, improved syntax highlighting, support for Razor formatting, better diagnostics, and features like "Go to Definition" and "Find All References" for Razor components and their parameters. |
+| **Support** | Still supported for maintaining existing applications. New development typically done using newer versions of Razor in ASP.NET Core. | N/A | N/A |
+| **Configuration** | N/A | To switch to the legacy editor, go to "Tools" > "Options" > "Environment" > "Preview Features" and uncheck the box that says "Use the new Razor editor for .NET Core apps". | To switch to the new editor, go to "Tools" > "Options" > "Environment" > "Preview Features" and check the box that says "Use the new Razor editor for .NET Core apps". |
```

Comment:
The option called out here is only available to internal Microsoft employees. It's worth either mentioning that, or changing this to mention the other location for flipping this flag, which is Tools > Options > Text Editor > HTML > Advanced